src/Symfony/Component/Config/Resource/ClassExistenceResource.php

```diff
@@ -121,20 +130,38 @@ public function unserialize($serialized)
      *
      * @internal
      */
-    public static function throwOnRequiredClass($class)
+    public static function _throwOnRequiredClass($class)
```

Comment:
can't we get rid of this one, by doing the check below only when $previous === null?

Reply:
Yes we can.

However, splitting responsabilities in 2 methods looks better for me :

`_throwOnRequiredClass` is a dedicated method for the `ClassExistenceResource` class.

`throwOnRequiredClass` can be used from anywhere. Having a check on an unrelated static property in this method doesn't help to understand the system.

Waiting for your final decision.

Reply:
Then we need two clearly named function - an underscore is not enough.
It'd also we could get rid of the try/catch around the method in warmers, doable?

Reply:
> an underscore is not enough.
`loadRequiredClass` (with a comment that it's an autoload function) & `throwOnRequiredClass` ?

> It'd also we could get rid of the try/catch around the method in warmers, doable?
Technically doable but that would mean `throwOnRequiredClass` wouldn't always throw the `\ReflectionException` (to silence it in the case it comes from a warmer) + it would need to rethrow passed previous exception. It gives this method two new added behaviors depending on the passed arguments while it's already kind of complicated. From my point of view, it's easier to understand with the try catch blocks in the warmer.

Reply:
@nicolas-grekas Just repushed. I did not change the names as they are just the best I could find 😕 To make things clearer, I added detailed comments. IMHO, since there are internal methods, it should be OK. Otherwise, waiting for naming help here.

Reply:
the second method could be named `dontThrowOnRequiredClass`, and with that name, we would remove the try/catch boilerplate

Reply:
Looking at this again.

Just realized that it needs to throw because there might be something done in the `\ReflectionException $e` catch block and in this case we want to catch the thrown `\ReflectionException` by the `throwOnRequiredClass` to reapply the same logic (cf what I did in the `ClassExistenceResource` with the goto).

However, I was able to reduce the try/catch boilerplate by rethrowing the previous exception directly in `throwOnRequiredClass` and by reusing a common method for warmers.